src/node/rpc/jwt_management.h

```diff
@@ -21,18 +22,62 @@
 
 namespace ccf
 {
+  static bool check_issuer(
+    const std::string& issuer, const std::string& constraint)
+  {
+    const auto issuer_domain = http::parse_url_full(issuer).host;
+    const auto constraint_domain = http::parse_url_full(constraint).host;
+
+    if (constraint_domain.empty())
+    {
+      return false;
+    }
+
+    // Either constraint's domain == issuer's domain or it is a subdomain, e.g.:
+    // limited.facebok.com
+    //        .facebok.com
+    if (issuer_domain != constraint_domain)
+    {
+      const auto start_seek =
+        issuer_domain.size() - (constraint_domain.size() + 1);
+      const auto count_seek = constraint_domain.size() + 1;
+      const auto pattern = "." + constraint_domain;
+
+      return start_seek > 0 // at least one letter preceeds .issuer.domain
+        && issuer_domain.substr(start_seek, count_seek) == pattern;
+    }
+
+    return true;
+  }
+
   static void remove_jwt_public_signing_keys(kv::Tx& tx, std::string issuer)
   {
-    auto keys = tx.rw<JwtPublicSigningKeys>(Tables::JWT_PUBLIC_SIGNING_KEYS);
-    auto key_issuer =
-      tx.rw<JwtPublicSigningKeyIssuer>(Tables::JWT_PUBLIC_SIGNING_KEY_ISSUER);
+    auto keys =
```

Comment:
I think we should keep the original code as well here, to remove entries from the `Legacy` tables.

Reply:
Cool, I understand and agree now - not worth peppering any `remove()` calls to legacy tables throughout the current code, because we don't want to try and precisely match the old write semantics. Instead we will deal with the old legacy state by adding a `clear()` call to the legacy tables, in a future release.

Reply:
Talked in person, decided to remove the keys from both revision when app calls "remove" explicitly, but don't touch old table on (re)setting new keys for an issuer.

Reply:
Ticket filed
https://github.com/microsoft/CCF/issues/6222